indexer/feature_meta.cpp

```diff
@@ -23,18 +23,23 @@ string Metadata::ToWikiURL(std::string v)
   if (colon == string::npos)
     return v;
 
-  // Spaces and % sign should be replaced in urls.
-  replace(v.begin() + colon, v.end(), ' ', '_');
-  string::size_type percent, pos = colon;
-  string const escapedPercent("%25");
-  while ((percent = v.find('%', pos)) != string::npos)
+  // Spaces, % and ? characters should be corrected to form a valid URL's path.
+  for (auto i = colon; i < v.size(); ++i)
   {
-    v.replace(percent, 1, escapedPercent);
-    pos = percent + escapedPercent.size();
+    auto & c = v[i];
+    if (c == ' ')
+      c = '_';
+    else if (c == '%')
+      v.insert(i + 1, "25");  // % => %25
+    else if (c == '?')
+    {
+      c = '%';
+      v.insert(i + 1, "3F");  // ? => %3F
+    }
```

Comment:
There is a disadvantage of using full percent-encoding. For example, https://it.wikipedia.org/wiki/Colosseo%20%28metropolitana%20di%20Roma%29 redirects to https://it.wikipedia.org/wiki/Colosseo_(metropolitana_di_Roma)

It means that there will be an unnecessary HTTP redirect. For a faster/better UX, it may be better to use a simpler approach, and only replace those characters which necessary. I can add an explicit comment about it.